encoding/codecv7.go

```diff
@@ -267,7 +276,7 @@ func (d *DACodecV7) CheckBatchCompressedDataCompatibility(b *Batch) (bool, error
 		return false, fmt.Errorf("failed to construct blob payload: %w", err)
 	}
 
-	_, compatible, err := d.checkCompressedDataCompatibility(payloadBytes)
+	_, compatible, err := d.checkCompressedDataCompatibility(payloadBytes, false /* checkLength */)
```

Comment:
Maybe we should add some comment to illustrate that why we should skip length checking here.

Reply:
makes sense. added in [feb63c9](https://github.com/scroll-tech/da-codec/pull/52/commits/feb63c9ee30a92f5f539603fb5c0d275ea290cb3).